src/Docker.PowerShell/Support/Win32ConsoleInterop.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace Docker.PowerShell.Support
+{
+    internal static class Win32ConsoleInterop
+    {
+        [DllImport("api-ms-win-core-file-l1-1-0.dll", EntryPoint = "CreateFileW", SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false)]
+        public static extern SafeFileHandle CreateFile(
+            string lpFileName,
+            int dwDesiredAccess,
+            System.IO.FileShare dwShareMode,
+            IntPtr securityAttrs,
+            System.IO.FileMode dwCreationDisposition,
+            int dwFlagsAndAttributes,
+            IntPtr hTemplateFile);
+
+        [DllImport("api-ms-win-core-console-l1-1-0.dll", SetLastError = true)]
+        public extern static bool GetConsoleMode(IntPtr handle, out int mode);
+
+        [DllImport("api-ms-win-core-console-l1-1-0.dll", SetLastError = true)]
+        public static extern bool SetConsoleMode(IntPtr handle, int mode);
+
+        [DllImport("api-ms-win-core-processenvironment-l1-1-0.dll", SetLastError = true)]
+        public static extern IntPtr GetStdHandle(int nStdHandle);
+
+        public const int STD_INPUT_HANDLE = -10;
+        public const int STD_OUTPUT_HANDLE = -11;
+        public const int STD_ERROR_HANDLE = -12;
+
+        public const int GENERIC_READ = unchecked((int)0x80000000);
```

Comment:
0x80000000 overflows int, so the unchecked tells it it's OK to do the cast anyway and make this -1. I could have used uint, I guess. I just copied the pinvoke definitions from coreclr.